### PR TITLE
Bugfix: min cg floating nodes

### PR DESCRIPTION
**Pull request by matteo-causalens**
## Description and Motivation
- Fixed a bug in `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph.get_minimal_graph` where the method was not adding the floating nodes correctly and it would return an empty graph. This also impacted the graph extension.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactor (moving code around, general refactoring improvements)
- [ ] Documentation (documentation)

## PR Checklist:

- [ ] I have reviewed my own PR? (review the PR as you are reviewing someone else's PR)
- [ ] I have implemented all requirements? (see JIRA, project documentation)
- [ ] I am affecting someone else's work? If so: I have added those people as reviewers?
- [ ] I have added comments to all the bits that are hard to follow
- [ ] At a bare minimum, I tested this manually
- [ ] I have added unit test(s)
- [ ] Is this a bugfix? If so have I added a regression test?
- [ ] Does this PR satisfy the [one PR, one concern](https://medium.com/@fagnerbrack/one-pull-request-one-concern-e84a27dfe9f1) rule?
- [ ] If needed, documentation has been added/updated?
- [ ] I have updated the appropriate changelog with a line for my changes
- [ ] If this PR breaks reproducibility, have I added a note in the changelog explaining the break in reproducibility
      and its extent?

### CODE REVIEW

File: cai_causal_graph/time_series_causal_graph.py [hunk 1]
```diff
@@ -432,6 +440,15 @@ def extend_graph(
             maxlag = minimal_graph.max_backward_lag
             assert maxlag is not None
 
+            # create all the nodes from 0 to backward_steps
```
Comment (steven-causalens) [inline cai_causal_graph/time_series_causal_graph.py:443]: Why is this change needed? Was this a bug too? It doesn't seem to be picked up in the changelog, so I would recommend to comment on it

**Review by steven-causalens — APPROVED** [review]
Looks good to me! I only skimmed the PR though. One small comment about the changelog

File: cai_causal_graph/time_series_causal_graph.py (hunk 1 above)
Comment (matteo-causalens) [reply cai_causal_graph/time_series_causal_graph.py]: No, I don't need it. It was a testing leftover. Reverted.

File: changelog.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## NEXT
+
+- Fixed a bug in `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph.get_minimal_graph` where the method
+  was not adding the floating nodes correctly. This also impacted the graph extension.
```
Comment (andrew-causalens) [inline changelog.md:6]: Have not had a chance to review full PR, but is there also a bug in `extend_graph` given our Slack discussion? Or does this fix also correct that?
```
ts = TimeSeriesCausalGraph()
ts.add_node('tier_0_0 lag(n=1)')
Out[13]: TimeSeriesNode("tier_0_0 lag(n=1)")
ts.extend_graph(1,0)
Out[14]: 
TimeSeriesCausalGraph(num_nodes=0, num_edges=0, id=-7856212696600437783)
Nodes: []
Edges: []
```

Comment (matteo-causalens) [reply changelog.md:6]: It was due to just the minimal graph. I am double checking with the other PR though to make sure everything is consistent.

**Review by andrew-causalens — APPROVED** [review]
